plugins/generator-1.19.4/forge-1.19.4/templates/gui/gui_window.java.ftl

```diff
@@ -0,0 +1,275 @@
+			{
+				{
+					setSuggestion(Component.translatable("gui.${modid}.${registryname}.${component.getName()}").getString());
+				}
+
+				@Override public void insertText(String text) {
+					super.insertText(text);
+
+					if (getValue().isEmpty())
+						setSuggestion(Component.translatable("gui.${modid}.${registryname}.${component.getName()}").getString());
+					else
+						setSuggestion(null);
+				}
+
+				@Override public void moveCursorTo(int pos) {
+					super.moveCursorTo(pos);
+
+					if (getValue().isEmpty())
+						setSuggestion(Component.translatable("gui.${modid}.${registryname}.${component.getName()}").getString());
+					else
+						setSuggestion(null);
+				}
+			}
+			</#if>;
+			${component.getName()}.setMaxLength(32767);
+
+			guistate.put("text:${component.getName()}", ${component.getName()});
+			this.addWidget(this.${component.getName()});
+		</#list>
+
+		<#assign btid = 0>
+
+		<#list data.getComponentsOfType("Button") as component>
+			${component.getName()} = new Button(new Button.Builder(Component.translatable("gui.${modid}.${registryname}${component.getName()}"), <@buttonOnClick component/>)
+				.bounds(this.leftPos + ${(component.x - mx/2)?int}, this.topPos + ${(component.y - my/2)?int}, ${component.width}, ${component.height}))
+			<#if !hasProcedure(buttonDisplayCondition)>
+			{}
+			<#else>
+			<@buttonDisplayCondition component/>
+			</#if>;
+
+			guistate.put("button:${component.getName()}", ${component.getName()});
+			this.addRenderableWidget(${component.getName()});
+
+			<#assign btid +=1>
+		</#list>
+
+		<#list data.getComponentsOfType("ImageButton") as component>
+		    ${component.getName()} = new ImageButton(
+				this.leftPos + ${(component.x - mx/2)?int}, this.topPos + ${(component.y - my/2)?int},
+            	${component.getWidth(w.getWorkspace())}, ${component.getHeight(w.getWorkspace())},
+				0, 0, ${component.getHeight(w.getWorkspace())},
+            	new ResourceLocation("${modid}:textures/screens/atlas/${component.getName()}.png"),
+            	${component.getWidth(w.getWorkspace())},
+				${component.getHeight(w.getWorkspace()) * 2},
+				<@buttonOnClick component/>
+			)<@buttonDisplayCondition component/>;
+
+			guistate.put("button:${component.getName()}", ${component.getName()});
+			this.addRenderableWidget(${component.getName()});
+
+			<#assign btid +=1>
+		</#list>
+
+		<#list data.getComponentsOfType("Checkbox") as component>
+			${component.getName()} = new Checkbox(this.leftPos + ${(component.x - mx/2)?int}, this.topPos + ${(component.y - my/2)?int},
+					20, 20, Component.translatable("gui.${modid}.${registryname}.${component.getName()}"), <#if hasProcedure(component.isCheckedProcedure)>
+				<@procedureOBJToConditionCode component.isCheckedProcedure/><#else>false</#if>);
+
+			guistate.put("checkbox:${component.getName()}", ${component.getName()});
+			this.addRenderableWidget(${component.getName()});
+		</#list>
+		
+		this.setInitialFocus(this.getFocused());

```

Comment:
This defines the component to be initially focused. The call as it is now does not make sense as it sets focus to currently focused, which is null, so the same as if you don't call this